src/io/flutter/run/bazel/BazelFields.java

```diff
@@ -10,75 +10,63 @@
 import com.intellij.execution.configurations.GeneralCommandLine;
 import com.intellij.execution.configurations.RuntimeConfigurationError;
 import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.InvalidDataException;
 import com.intellij.openapi.util.io.FileUtil;
 import com.intellij.openapi.util.text.StringUtil;
 import com.intellij.openapi.vfs.CharsetToolkit;
 import com.intellij.openapi.vfs.LocalFileSystem;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.jetbrains.lang.dart.sdk.DartConfigurable;
 import com.jetbrains.lang.dart.sdk.DartSdk;
+import com.sun.org.apache.xpath.internal.operations.Bool;
 import io.flutter.FlutterBundle;
 import io.flutter.bazel.Workspace;
-import io.flutter.bazel.WorkspaceCache;
 import io.flutter.dart.DartPlugin;
 import io.flutter.run.MainFile;
+import io.flutter.run.bazelTest.BazelTestFields;
 import io.flutter.run.daemon.FlutterDevice;
 import io.flutter.run.daemon.RunMode;
+import io.flutter.utils.ElementIO;
+import org.jdom.Element;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Map;
+
 import static io.flutter.run.daemon.RunMode.*;
 
 /**
  * The fields in a Bazel run configuration.
  */
 public class BazelFields {
 
-  private @Nullable String entryFile;
-  private @Nullable String launchScript;
-  private @Nullable String bazelTarget;
-  private boolean enableReleaseMode = false;
-  private @Nullable String additionalArgs;
+  @Nullable
+  private final String bazelTarget;
+
+  private final boolean enableReleaseMode;
+
+  @Nullable
+  private final String bazelArgs;
```

Comment:
I've opted to make the bazel fields all immutable to make it easier to understand state changes in the run configurations.

Is there a way I can annotate the class as immutable to enforce this for future contributors?

Reply:
Short answer: No, simply document it.

Right answer: looking through the Dart Plugin, there is one instance of `@Immutable` I found, but it was imported via `thirdPartySrc/`, I don't believe that anything is enforced in the JVM runtime. Again documentation is the best option here.

Reply:
Noted.  I've documented this.